container.go

```diff
@@ -112,6 +145,8 @@ func (c *Container) Resolve(obj interface{}) (err error) {
 	objElemType := reflect.TypeOf(obj).Elem()
 	objVal := reflect.ValueOf(obj)
 
+	c.Lock()
+	defer c.Unlock()
```

Comment:
might also be good to just put the RLock around the map access, but not a big deal.